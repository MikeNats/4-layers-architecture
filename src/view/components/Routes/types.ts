export type RoutesProps = {
    authenticated: boolean
  }
