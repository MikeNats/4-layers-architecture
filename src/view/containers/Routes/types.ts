import { LogInReducer } from '../../../store/state/auth/reducers/types'

export type RoutesProps = {
    authenticated: boolean
  }
export type RoutesMapStateToProps = {
  auth: LogInReducer
}