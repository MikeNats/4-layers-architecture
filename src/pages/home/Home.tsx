import React from 'react';
import { connect } from 'react-redux'
import { Dispatch } from 'redux'
import { fetchtransactions } from '../../actions'
import { TransactionPayLoadItemType } from '../../actions/transactions/types'

type HomePropsType = {
  payload: TransactionPayLoadItemType[]
  isFetching: Boolean,
  didInvalidate: Boolean,
  dispatch: Dispatch<any>
};

type HomeStateType = {
  transactions: {
    payload: TransactionPayLoadItemType[]
    isFetching: Boolean,
    didInvalidate: Boolean,
    errorCode: number
  }
}

class Home extends React.Component <HomePropsType, HomeStateType> {
  componentDidMount() {
    const { dispatch } = this.props;
    fetchtransactions(1)(dispatch);
  }
 
  render() {
    return (<section>
    <h1>{this.props.isFetching}</h1>
      <section>
      </section>
      <aside>
      </aside>
    </section>);
  }
}

const mapStateToProps = (state: HomeStateType) => ({
  payload: state.transactions.payload,
  isFetching: state.transactions.isFetching,
  didInvalidate: state.transactions.didInvalidate,
  errorCode: state.transactions.errorCode,
})

export default connect(mapStateToProps)(Home);