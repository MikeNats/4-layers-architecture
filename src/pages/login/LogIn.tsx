import React from 'react';

export default () => {
  return (
    <section>
      <h1>log in</h1>
      <section>
      </section>
      <aside>
      </aside>
    </section> 
  );
}