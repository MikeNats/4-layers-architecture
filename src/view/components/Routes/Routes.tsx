import React, { Suspense, lazy } from 'react';
import AuthPage from '../../pages/AuthPage/AuthPage'
import ErrorPage from '../../pages/ErrorPage/ErrorPage'; 
import Page from  './Page/Page'
import { PATHS } from '../../../enums' 
import { RoutesProps } from './types'
import {
  Switch,
  Redirect,
  HashRouter as Router, 
} from 'react-router-dom';
const TransactionsPage = lazy(() => import('../../pages/TransactionsPage/TransactionsPage'));

const Routes = ({authenticated} : RoutesProps) => 
  <Router>
    <Suspense fallback={<div>Loading...</div>}>
      <Switch>
        <Page exact path={PATHS.LOGIN} component={AuthPage} /> 
        <Page exact path="/" component={TransactionsPage} protectedRoute authenticated={authenticated} hasHeaderFooter/>
        <Page exact path={PATHS.ERROR} component={ErrorPage}/> 
        <Redirect to="/" />
      </Switch>
    </Suspense>
  </Router> 

export default Routes

 