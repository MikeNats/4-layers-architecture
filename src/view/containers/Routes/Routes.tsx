import React, { Suspense, lazy } from 'react';
import Auth from '../Auth/Auth'
import ErrorView from '../../presentational/ErrorView/ErrorView'; 
import PrivateRoute from  '../../presentational/common/PrivateRoute/PrivateRoute'
import PATHS from './PATHS' 
import {connect} from 'react-redux';
import { RoutesProps, RoutesMapStateToProps } from './types'
import {
  Switch,
  Route,
  Redirect,
  HashRouter as Router, 
} from 'react-router-dom';
const Transactions = lazy(() => import('../Transactions/Transactions'));

export class Routes extends React.Component<RoutesProps, {}>{

  render(){
    return (<Router>
      <Suspense fallback={<div>Loading...</div>}>
        <Switch>
          <Route exact path={PATHS.LOGIN} component={Auth} /> 
          <PrivateRoute exact={true} path="/" auth={this.props.authenticated} component={Transactions} />
          <Route exact={true} path={PATHS.ERROR} component={ErrorView}/> 
          <Redirect to="/" />
        </Switch>
      </Suspense>
  </Router>)
  }
}

const mapStateToProps = (state: RoutesMapStateToProps) => ({
  authenticated: state.auth.authenticated,
})

export default connect(mapStateToProps)(Routes)

 