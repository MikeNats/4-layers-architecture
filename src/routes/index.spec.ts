import LogIn from '../pages/login/LogIn'
import Routes from '.'
import Home from '../pages/home/Home'
import { shallow } from 'enzyme';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
Enzyme.configure({ adapter: new Adapter() })


describe("ApRoutes", () => {
  describe('component', () => {

    let element: JSX.Element

    beforeEach(() => {
      element = Routes()
    });

    it('renders as expected', () => {
      const component = shallow(element);
      expect(component).toMatchSnapshot();
    });

    it('route / to Home', () => {
      const component = shallow(element);
      expect(component.find('Route[path="/"]').first().prop('component')).toBe(Home);
    });

    it('route /login to LogIn', () => {
      const component = shallow(element);
      expect(component.find('Route[path="/login"]').first().prop('component')).toBe(LogIn);
    });
  });

});