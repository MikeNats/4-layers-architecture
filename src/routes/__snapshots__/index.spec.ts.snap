// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ApRoutes component renders as expected 1`] = `ShallowWrapper {}`;
